runtime/include/atomics/cstdlib-intrinsics/chpl-atomics.h

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2004-2019 Cray Inc.
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "chpl-comp-detect-macros.h"
+
+/* This is a portability shim that allows us to use cstdlib atomics if the
+ * compiler supports them, and falls back to intrinsics if it doesn't. This
+ * allows us to build our module with a version that doesn't support cstdlib
+ * atomics, but still allows the generated code to take advantage of them if
+ * the user has a new enough compiler. */
+
+// GCC >= 5.0
+#if RT_COMP_CC == RT_COMP_GCC
+  #if RT_COMP_GCC_VERSION_MAJOR >= 5 && !defined(__STDC_NO_ATOMICS__)
```

Comment:
Let's consider factoring out the check for __STDC_NO_ATOMICS__ since arguably we should never set USE_CSTDLIB in that case, regardless of the compiler